src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -247,6 +251,37 @@ public String relativePath(VirtualFile rootParent) {
         return result;
     }
 
+    /**
+     * Returns the root directory path of the current virtual file.
+     * <p>
+     * If the virtual file belongs to a known space, it is used as root, otherwise <tt>/</tt> is returned.
+     *
+     * @return the root directory path
+     */
+    @Nullable
+    public VirtualFile root() {
+        List<VirtualFile> fileList = pathList();
+        if (fileList.isEmpty()) {
+            return null;
+        }
+
+        if (fileList.size() > 1 && objectStorage.isKnown(fileList.get(1).name())) {
+            return fileList.get(1);
+        }
+
+        return fileList.get(0);
+    }
+
+    /**
+     * Indicates if the root directory is considered the working directory.
+     *
+     * @return <tt>true</tt> when the root is the working directory, <tt>false</tt> otherwise
+     * @see ObjectStorage#WORK_SPACE_NAME
+     */
+    public boolean isRootWorkDirectory() {
```

Comment:
the VirtualFile already has quite a borad API- Shouldn't we move this either into L3Uplink or into a helper in the utils package? We then could also check if this is a wrapper as generated by L3Uplink and extract the space name there - and also provide a constant / another utility method for work...